content/en/docs/sdk-utilities/s3.md

```diff
@@ -101,6 +101,48 @@ before sending it to {{% alias service="S3" %}}. `Uploader` calculates the expec
 file to {{% alias service="S3" %}}. If the current value of `PartSize` requires more than 10,000 parts to upload the
 file, `Uploader` increases the part size value so that fewer parts are required.
 
+### S3 Object Integrity Features (checksum and md5)
```

Comment:
I think this is overall focused a bit too much on the feature itself, instead I think we want to just document the differences/behavior w.r.t `ContentMD5/Checksum*/ChecksumAlgorithm` settings. Perhaps we should work this out in a quip doc first rather than going back and forth in PR but here is a first stab at it:


```
S3 [Object Integrity Checks](https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html) in {{% alias service="S3" %}} can be used with the transfer manager `Upload` function.

\```go
result, err := uploader.Upload(context.TODO(), &s3.PutObjectInput{
	Bucket: aws.String("my-bucket"),
	Key:    aws.String("my-object-key"),
	Body:   uploadFile,
	ChecksumAlgorithm: types.ChecksumAlgorithmSha256,
})
\```

There are some differences from `PutObject` and `UploadPart` to be aware of. It is not recommended to set explicit `ContentMD5` or any of the `Checksum*` values (e.g. `ChecksumCRC32`, `ChecksumCRC32C`, `ChecksumSHA1`, or `ChecksumSHA256`) directly when using the `Upload` function. This is because the `Uploader` will internally convert large upload requests to a multipart upload for performance. The provided checksum values would be invalid for individual parts and thus are ignored by the SDK. The recommended way to enable object integrity checks is to set a `ChecksumAlgorithm` which will be used for single and multi-part uploads. 

```

Reply:
I share a quip containing the draft https://quip-amazon.com/1d1aAwaSz3ya/Dev-guide-draft-for-S3-object-integrity-check